docs/codelabs/github_actions.md

```diff
@@ -0,0 +1,177 @@
+
+If you really get
+stuck you can find us on [gitter](https://gitter.im/please-build/Lobby)!
+
+## GitHub Actions
+Duration: 5
+
+GitHub Actions is an extensible CI/CD platform provided by GitHub.
+Compared to other CI/CD solutions, GitHub Actions allows you to build all kinds automations (called workflows) triggered by various events (eg. pushing code to a branch).
+
+### Setting up GitHub Actions
+
+Workflow definitions are simple YAML files stored in the `.github/workflows` directory of your repository.
+
+The following snippet triggers a workflow named `CI` whenever commits are pushed to the `master` branch:
+
+```yaml
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Test
+        run: echo "Tests passed"
+```
+
+Go ahead and add the above snippet to `.github/workflows/ci.yaml` in your project. Then go to `https://github.com/YOU/YOUR-PROJECT/actions` and observe the workflow.
+
+## Please build
+Duration: 4
+
+Now we have a project setup with GitHub Actions, it's time to start building with Please! Let's change `ci.yaml` a little:
+
+```yaml
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+
+    steps:
+      # Setup your language of choice here:
+      # https://github.com/actions/?q=setup-&type=&language=
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      # Run please build
+      - name: Test
+        run: ./pleasew build //...
+```
+
+Compared to the example earlier, this workflow uses the `pleasew` script to download Please and build the project.
+
+Notice the `//...` bit at the end of the command: it's necessary on GitHub Actions.
+Check [this](https://github.com/thought-machine/please/issues/1174) issue for more details.
+
+## `setup-please` action
+Duration: 10
+
+The [setup-please](https://github.com/sagikazarmark/setup-please-action) action provides better integration for Please.
+
+### What is an _action_?
+
+As you've seen in the previous examples, workflows consist of _steps_.
+A workflow step can be as simple as a shell script:
+
+```yaml
+- name: Test
+  run: ./pleasew build //...
+```
+
+Shell scripts (no matter how awesome they are) are not always the right tool for the job. Complex build steps might require a more expressive language which takes us to the second type of workflow steps, called _actions_:
+
+```yaml
+- name: Checkout code
+  uses: actions/checkout@v2
+```
+
+An _action_ can be written in any language (distributed as Docker images), but JavaScript is supported natively.
+
+### Why not just use `./pleasew`?
```

Comment:
This doesn't render to HTML very well with the backticks.

Reply:
Should I just remove it? I left it there because it still makes sense in markdown.

Reply:
Yeah I think the website is more important than the markdown.

Reply:
Changed